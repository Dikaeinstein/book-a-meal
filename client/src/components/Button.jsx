import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, className }) => (
  <button className={className}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
