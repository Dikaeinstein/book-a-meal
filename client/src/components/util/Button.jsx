import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, ...otherProps }) => (
  <button {...otherProps}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
