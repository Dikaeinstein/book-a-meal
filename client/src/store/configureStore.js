import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
