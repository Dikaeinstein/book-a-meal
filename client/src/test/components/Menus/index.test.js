import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import sweetalert from 'sweetalert';
import ConnectedMenus, { Menus } from '../../../components/Menus';
import Button from '../../../components/util/Button';

jest.mock('sweetalert');

describe('<Menus />', () => {
  const props = {
    userRole: 'superAdmin',
    isSet: false,
    fetchMenuError: null,
  };
  it('should call handleCloseModal when modal is closed', () => {
    const closeModalSpy = jest.spyOn(Menus.prototype, 'handleCloseModal');
    const wrapper = shallow(<Menus {...props} />);
    const closeModal = wrapper.find(Button).last();
    closeModal.simulate('click');
    expect(closeModalSpy).toHaveBeenCalled();
    expect(wrapper.state().isOpen).toBeFalsy();
  });
  it('should call handleOpenModal when set menu is clicked', () => {
    const openModalSpy = jest.spyOn(Menus.prototype, 'handleOpenModal');
    sweetalert.mockResolvedValue(Promise.resolve());
    let wrapper = shallow(<Menus {...props} />);
    let setMenu = wrapper.find(Button).first();
    setMenu.simulate('click');
    expect(openModalSpy).toHaveBeenCalled();
    expect(wrapper.state().isOpen).toBeTruthy();
    wrapper = shallow(<Menus {...{ ...props, userRole: 'caterer' }} />);
    setMenu = wrapper.find(Button).first();
    setMenu.simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
  it('should call handleMenuUpdate when updateMenu is clicked', () => {
    const openModalSpy = jest.spyOn(Menus.prototype, 'handleMenuUpdate');
    sweetalert.mockResolvedValue(Promise.resolve());
    let wrapper = shallow(<Menus {...props} />);
    let updateMenu = wrapper.find(Button).at(1);
    updateMenu.simulate('click');
    expect(openModalSpy).toHaveBeenCalled();
    sweetalert.mockResolvedValue(Promise.resolve());
    wrapper = shallow(<Menus {...{ ...props, userRole: 'caterer' }} />);
    updateMenu = wrapper.find(Button).at(1);
    updateMenu.simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with correct props from redux', () => {
    const mockStore = configureStore();
    const store = mockStore({
      menu: { isSet: false, fetchError: null },
      user: { data: { role: 'caterer' } },
    });
    const wrapper = shallow(<ConnectedMenus store={store} />);
    expect(wrapper.props().isSet).toBeFalsy();
    expect(wrapper.props().fetchMenuError).toBeNull();
    expect(wrapper.props().userRole).toEqual('caterer');
  });
  it('should match snapshot', () => {
    const tree = shallow(<Menus {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
