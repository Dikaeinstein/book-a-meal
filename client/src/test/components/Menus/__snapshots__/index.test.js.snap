// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menus /> should call handleMenuUpdate when updateMenu is clicked 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "#f0f1f2",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "1rem 4rem 0 4rem",
      }
    }
  >
    <Button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "margin": "1rem 1rem 1rem 0",
        }
      }
      value="Set Up Menu"
    />
    <Button
      className="btn btn-default"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "margin": "1rem 0 1rem 0",
        }
      }
      value="Update Menu"
    />
  </div>
  <main
    className="bg-light main"
    style={
      Object {
        "minHeight": "calc(100vh - 240px)",
      }
    }
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      contentLabel="Menu"
      isOpen={false}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "bottom": "auto",
            "left": "50%",
            "marginRight": "-50%",
            "minWidth": "20rem",
            "right": "auto",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "width": "60%",
          },
          "overlay": Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        }
      }
    >
      <Button
        className="close"
        onClick={[Function]}
        value="×"
      />
      <ErrorWrapper
        closeModal={[Function]}
      />
    </Modal>
    <ErrorWrapper
      error={null}
    />
  </main>
  <Footer />
</div>
`;

exports[`<Menus /> should call handleOpenModal when set menu is clicked 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "#f0f1f2",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "1rem 4rem 0 4rem",
      }
    }
  >
    <Button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "margin": "1rem 1rem 1rem 0",
        }
      }
      value="Set Up Menu"
    />
    <Button
      className="btn btn-default"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "margin": "1rem 0 1rem 0",
        }
      }
      value="Update Menu"
    />
  </div>
  <main
    className="bg-light main"
    style={
      Object {
        "minHeight": "calc(100vh - 240px)",
      }
    }
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      contentLabel="Menu"
      isOpen={false}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "bottom": "auto",
            "left": "50%",
            "marginRight": "-50%",
            "minWidth": "20rem",
            "right": "auto",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "width": "60%",
          },
          "overlay": Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        }
      }
    >
      <Button
        className="close"
        onClick={[Function]}
        value="×"
      />
      <ErrorWrapper
        closeModal={[Function]}
      />
    </Modal>
    <ErrorWrapper
      error={null}
    />
  </main>
  <Footer />
</div>
`;

exports[`<Menus /> should match snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "#f0f1f2",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "1rem 4rem 0 4rem",
      }
    }
  >
    <Button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "margin": "1rem 1rem 1rem 0",
        }
      }
      value="Set Up Menu"
    />
    <Button
      className="btn btn-default"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "margin": "1rem 0 1rem 0",
        }
      }
      value="Update Menu"
    />
  </div>
  <main
    className="bg-light main"
    style={
      Object {
        "minHeight": "calc(100vh - 240px)",
      }
    }
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      contentLabel="Menu"
      isOpen={false}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "bottom": "auto",
            "left": "50%",
            "marginRight": "-50%",
            "minWidth": "20rem",
            "right": "auto",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "width": "60%",
          },
          "overlay": Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        }
      }
    >
      <Button
        className="close"
        onClick={[Function]}
        value="×"
      />
      <ErrorWrapper
        closeModal={[Function]}
      />
    </Modal>
    <ErrorWrapper
      error={null}
    />
  </main>
  <Footer />
</div>
`;
